File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/data/isyncable_data_source.dart
abstract class ISyncableDataSource<T> {
  Future<List<T>> addMany(List<T> entities);
  Future<T> add(T entity);
  Future<void> hardDelete();
  Future<List<T>> updateMany(List<T> entities);
  Future<List> syncUpdate(List entities);
  Future<T> update(String id, T entity);
  Future<List<T>> getAll();
  Future<T?> findById(String id);
  Future<List<T>> findByIds(List<String> ids);
  Future<int> count();
  Future<void> dispose();
}

File: fast-sync-typescript-client-sdk/packages/fast-sync-client/src/abstraction/data/isyncable_data_source.ts
export interface ISyncableDataSource<T> {
  addMany(entities: T[]): Promise<T[]>;
  add(entity: T): Promise<T>;
  hardDelete(): Promise<void>;
  updateMany(entities: T[]): Promise<T[]>;
  syncUpdate(entities: T[]): Promise<T[]>;
  update(id: string, entity: T): Promise<T>;
  getAll(): Promise<T[]>;
  findById(id: string): Promise<T | null>;
  findByIds(ids: string[]): Promise<T[]>;
  count(): Promise<number>;
  dispose(): Promise<void>;
}
